packages/ai/core/tool/mcp/client.ts

```diff
@@ -0,0 +1,341 @@
+import { z, ZodType } from 'zod';
+import { jsonSchema } from '@ai-sdk/ui-utils';
+import { JSONSchema7 } from '@ai-sdk/provider';
+import { MCPClientError } from '../../../errors';
+import { inferParameters, tool, ToolExecutionOptions } from '../tool';
+import { Tool } from '../tool';
+import {
+  CallToolRequest,
+  CallToolResult,
+  CallToolResultSchema,
+  Configuration as ClientConfiguration,
+  InitializeResultSchema,
+  JSONRPCError,
+  JSONRPCNotification,
+  JSONRPCRequest,
+  JSONRPCResponse,
+  LATEST_PROTOCOL_VERSION,
+  ListToolsRequest,
+  ListToolsResult,
+  ListToolsResultSchema,
+  MCPTransport,
+  Notification,
+  Request,
+  RequestOptions,
+  SUPPORTED_PROTOCOL_VERSIONS,
+  TransportConfig,
+  McpToolSet,
+  ToolSchemas,
+} from './types';
+import { createMcpTransport } from './transport';
+
+interface MCPClientConfig {
+  /** Transport configuration for connecting to the MCP server */
+  transport: TransportConfig;
+  /** Optional client name, defaults to 'ai-sdk-mcp-client' */
+  name?: string;
+  /** Optional client version, defaults to '1.0.0' */
+  version?: string;
```

Comment:
hm the client version is what we support, right? so should the user be able to specify it?

Reply:
client config is more informational, not really used in a critical capacity (e.g. compatibility checking b/w client and server) - i'd like to advocate for keeping it so that users can debug/track specific clients they create if necessary!

Reply:
we may want this internally, agreed - the reason i bring this up is that i'm not convinced we should expose it to our users